routes/web.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+use Illuminate\Support\Facades\Route;
+
+/*
+|--------------------------------------------------------------------------
+| Web Routes
+|--------------------------------------------------------------------------
+|
+| Here is where you can register web routes for your application. These
+| routes are loaded by the RouteServiceProvider within a group which
+| contains the "web" middleware group. Now create something great!
+|
+*/
+
+Route::get('/', function () {
+    return view('welcome');
+});
+
+Auth::routes();
+
+Route::get('/home', [App\Http\Controllers\HomeController::class, 'index'])->name('home');
+
+Route::get('roles/index',[ 'middleware' => 'App\Http\Middleware\RoleMiddleware:MANAGE_ROLES', 'uses' => 'App\Http\Controllers\RoleController@index']);
+Route::get('roles/create',[ 'middleware' => 'App\Http\Middleware\RoleMiddleware:MANAGE_ROLES', 'uses' => 'App\Http\Controllers\RoleController@create']);
+Route::get('roles/{id}/edit',[ 'middleware' => 'App\Http\Middleware\RoleMiddleware:MANAGE_ROLES', 'uses' => 'App\Http\Controllers\RoleController@edit']);
+Route::delete('roles/{id}',[ 'middleware' => 'App\Http\Middleware\RoleMiddleware:MANAGE_ROLES', 'uses' => 'App\Http\Controllers\RoleController@delete']);
```

Comment:
I assume here this is wrong? App\Http\Middleware\RoleMiddleware:<any option>

You wanted to reference this => App\Enums\UserPermissionsEnum::<one of the consts>

Reply:
I made a change here and I am using the alias role instead of App\Http\Middleware\RoleMiddleware. Then I need to pass the allowed permission that needed in order to navigate to the specified root. I tried to pass the value of the enum (i.e App\Enums\UserPermissionsEnum::MANAGE_ROLES) but instead of the value it pass the App\Enums\UserPermissionsEnum::MANAGE_ROLES as string.